### PR TITLE
feat(*): Add hiddenWord and show random letters

### PR DESCRIPTION
**Pull request by dhruvd-grappus**
Hide correct word for non drawing players

https://github.com/noorbhatia10/grappus_skribbl/assets/110366216/f0ab25b1-56d6-4a8e-adf7-fd63905aafd6

### CODE REVIEW

File: api/lib/session/bloc/session_bloc.dart [hunk 1]
```diff
@@ -207,6 +208,34 @@ class SessionBloc extends BroadcastBloc<SessionEvent, SessionState> {
...
+          state.hiddenAnswer,
+          charIndex,
+          state.correctAnswer[charIndex],
+        );
+
+        emit(
+          state.copyWith(
+            hiddenAnswer: hiddenAnswer,
+          ),
+        );
```
Comment (noorbhatia10) [inline api/lib/session/bloc/session_bloc.dart:235]: How are we ensuring that all letters will be shown when the timers ends? i.e. the logic to show number of letters at every mark according to the word length

Comment (dhruvd-grappus) [reply api/lib/session/bloc/session_bloc.dart:235]: I didnt understand,
we are showing some letters as a hint to the players
we dont need to show all letters, correct answer will be shown when the round is complete.

Comment (noorbhatia10) [reply api/lib/session/bloc/session_bloc.dart:235]: let's say we have a very long word and the timer is only of 60 seconds, in this case we need to show more than one letter at 50/40/30 mark.

Comment (dhruvd-grappus) [reply api/lib/session/bloc/session_bloc.dart:235]: Is this critical for our first prototype? we can include this in refinements for now

**Review by noorbhatia10 — APPROVED** [review]

**Pull request merged**